scripts/manageacl.py

```diff
@@ -0,0 +1,158 @@
+        await self.search_client.__aexit__(*args, **kwargs)
+        await self.search_index_client.__aexit__(*args, **kwargs)
+
+    def __init__(
+        self,
+        service_name: str,
+        index_name: str,
+        document: str,
+        acl_action: str,
+        acl_type: str,
+        acl: str,
+        credentials: AsyncTokenCredential | AzureKeyCredential,
+        verbose: bool = False,
+    ):
+        self.service_name = service_name
+        self.index_name = index_name
+        self.credentials = credentials
+        self.document = document
+        self.acl_action = acl_action
+        self.acl_type = acl_type
+        self.acl = acl
+        self.verbose = verbose
+
+    async def run(self):
+        if self.acl_action == "enable_acls":
+            if self.verbose:
+                print(f"Enabling acls for index {self.index_name}")
+            index_definition = await self.search_index_client.get_index(self.index_name)
+            if not any(field.name == "oids" for field in index_definition.fields):
+                index_definition.fields.append(
+                    SimpleField(
+                        name="oids", type=SearchFieldDataType.Collection(SearchFieldDataType.String), filterable=True
+                    )
+                )
+            if not any(field.name == "groups" for field in index_definition.fields):
+                index_definition.fields.append(
+                    SimpleField(
+                        name="groups", type=SearchFieldDataType.Collection(SearchFieldDataType.String), filterable=True
+                    )
+                )
+
+            await self.search_index_client.create_or_update_index(index_definition)
+            return
+
+        filter = f"sourcefile eq '{self.document}'"
+        result = await self.search_client.search(
+            "", filter=filter, select=["id", self.acl_type], include_total_count=True
+        )
+        if await result.get_count() == 0:
+            if args.verbose:
+                print(f"No documents match {self.document} - exiting")
+            return
+
+        documents_to_merge = []
+        async for document in result:
+            if self.acl_action == "view":
+                # Assumes the acls are consistent across all sections of the document
+                print(document[self.acl_type])
+                return
+
+            if self.acl_action == "remove":
+                new_acls = [acl_value for acl_value in document[self.acl_type] if acl_value != self.acl]
+            elif self.acl_action == "add":
+                new_acls = document[self.acl_type]
+                if not any(acl_value == self.acl for acl_value in new_acls):
+                    new_acls.append(self.acl)
+            else:
+                new_acls = []
+            documents_to_merge.append({"id": document["id"], self.acl_type: new_acls})
+
+        await self.search_client.merge_documents(documents=documents_to_merge)
+        if self.verbose:
```

Comment:
Ditto my comment re logging. Found nice example of how azure sdk handles it for scripts: https://github.com/Azure/azure-sdk-for-python/blob/6f168b61101bbf851a30c085747d51470db4be0c/tools/azure-sdk-tools/packaging_tools/generate_package.py#L46